### PR TITLE
ARROW-13729: [Website] setup datafusion python binding docs

### LINKED ISSUE #837: Release documentation for python binding
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Follow up from https://github.com/apache/arrow-datafusion/issues/771#issuecomment-888160309 and https://github.com/apache/arrow-datafusion/issues/771#issuecomment-888490358.

**Describe the solution you'd like**

We should have a python API and user doc similar to https://arrow.apache.org/docs/python.

**Describe alternatives you've considered**

**Additional context**

Discussion on how to host datafusion website: https://lists.apache.org/thread.html/r0ed76cc60cdf651e8cf5c82a21cc64114c1f6d174dc5487434bd32ef%40%3Cdev.arrow.apache.org%3E

### PR DESCRIPTION
**Pull request by jimexist**
closes https://github.com/apache/arrow-datafusion/issues/837

cc @houqp @alamb @jorgecarleitao and @andygrove 

Screen Shot 2021-08-24 at 16 25 13
Screen Shot 2021-08-24 at 16 26 01

### CODE REVIEW

**Comment by houqp** [comment]
Looks amazing, thanks @Jimexist !

 @alamb @jorgecarleitao and @andygrove, do we need to wait for a voted datafusion 5.1.0 release before we can merge this PR?

**Comment by alamb** [comment]
@houqp  I do not think any official vote is required to release documentation as I wouldn't personally consider it part of the release.

**Comment by houqp** [comment]
@Jimexist do we need to update https://github.com/apache/arrow/blob/master/docs/source/developers/documentation.rst as well to mention installation of the datafusion package?

I am not familiar with the automated doc build and publish pipeline, @kszucs @pitrou @lidavidm @wesm @fsaintjacques do we need to update automation to build and install the datafusion python package?

File: docs/source/conf.py [hunk 1]
```diff
@@ -37,14 +37,10 @@
...
+import datafusion
 import pyarrow
 
-
-sys.path.extend([
-    os.path.join(os.path.dirname(__file__),
-                 '..', '../..')
-
-])
+sys.path.extend([os.path.join(os.path.dirname(__file__), "..", "../..")])
```
Comment (pitrou) [inline docs/source/conf.py:43]: Most of this file is auto-generated. By making style changes we will make it more difficult to diff the current file against a pristine file.

**Comment by pitrou** [comment]
I don't think Datafusion is a "supported environment" at the same level as C++, Python, etc. Since this documents the Python Datafusion library, this could go into the "Python" section.

However, since Datafusion is a distinct library from PyArrow (and has its own release schedule), it seems to me it would be better if its documentation lived in a different place altogether. That could be on https://arrow.apache.org, but it could also be hosted on https://readthedocs.org/, for example.

@amol-  What do you think?

**Comment by amol-** [comment]
> I don't think Datafusion is a "supported environment" at the same level as C++, Python, etc. Since this documents the Python Datafusion library, it could go into the existing "Python" section.
> 
> However, since Datafusion is a distinct library from PyArrow (and has its own release schedule), it seems to me it would be better if its documentation lived in a different place altogether. That could be on https://arrow.apache.org, but it could also be hosted on https://readthedocs.org/, for example.
> 
> @amol- What do you think?

I think I would vote for having datafusion into its own documentation (on readthedocs or anywhere else as you suggested), surely it's not one of the supported environments, but in the end is not even part of libarrow/pyarrow itself so I'd avoid mixing its documentation into the one of pyarrow.

**Comment by jorgecarleitao** [comment]
Relevant mailing list discussion: https://lists.apache.org/thread.html/r0ed76cc60cdf651e8cf5c82a21cc64114c1f6d174dc5487434bd32ef%40%3Cdev.arrow.apache.org%3E

**Comment by houqp** [comment]
based on the last reply from @kszucs (https://lists.apache.org/thread.html/r5c9341fe5360ae249532724da2bf92bd2ed661d1c58f599212e82107%40%3Cdev.arrow.apache.org%3E), how about we reuse the current sphinx setup/theme for datafusion, but have a new self-contained website created under `/datafusion`  folder? From there, we could host not just the python api/user doc, but also datafusion user docs in general.

**Comment by houqp** [comment]
@pitrou @amol- based on @wesm's suggestion in https://lists.apache.org/thread.html/r9500355019f7e438ed2417bce577e1a76dbcba742c2c7e1008dcffd5%40%3Cdev.arrow.apache.org%3E, are you all cool with us hosting datafusion related docs in its own website under https://arrow.apache.org/datafusion?

**Comment by Jimexist** [comment]
@houqp thanks for migrating this code. maybe now I can close this pull request?

**Comment by houqp** [comment]
I think so, all of your change in this PR should have been merged into datafusion repo now :)

**Pull request closed without merging**